File: demo/functions/calc/divide-by/Handler.fs
namespace Calc
open Amazon.Lambda.Core

[<assembly:LambdaSerializer(typeof<Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer>)>]
do ()

[<CLIMutable>]
type Request = { _fal: string }
type ReqInput = { input: string }

type Response = { _fal: string }
type ResOutput = { mutable success: bool; mutable output: string }


module DivideBy =
    open System
    open System.IO
    open System.Text
    open FSharp.Json

    let run(request: Request) =
        printfn "Got event: %A" request

        let res: ResOutput = { success = false; output = null }

        try
            let fal = Json.deserialize<ReqInput> request._fal
            let input = int fal.input

            res.success <- true
            res.output <- string (input / 2)

        with :? System.FormatException as ex ->
            printfn "Could not perform operation: %s" ex.Message

        { _fal = Json.serializeU res }

File: demo/functions/calc/power-of/handler.py
import json
import pprint


def response(output):
    _fal = {
        'success': output is not None,
        'output': str(output or '')
    }

    return { '_fal': json.dumps(_fal, separators=(',', ':')) }


def power_of(event, context):
    print('Got event: ' + pprint.pformat(event))

    if not event.get('_fal'):
        return response(None)

    try:
        fal = json.loads(event['_fal'])
        fal_input = int(fal['input'])

        return response(pow(fal_input, 2))

    except (TypeError, ValueError) as err:
        print('Could not perform operation: ' + str(err))

    return response(None)
